packages/bs-swap/src/services/SimpleSwapService.ts

```diff
@@ -198,7 +198,10 @@ export class SimpleSwapService<BSName extends string = string> implements SwapSe
       blockchainServicesByName: this.#blockchainServicesByName,
       chainsByServiceName: this.#chainsByServiceName,
     })
-    this.#availableTokensToUse = { loading: false, value: tokens }
+
+    const filteredTokens = tokens.filter(token => token.blockchain && token.decimals && token.hash)
```

Comment:
Decimals can be zero, change for: `const filteredTokens = tokens.filter(token => token.blockchain && typeof token.decimals === 'number' && token.hash)`.